src/fileAttachment.js

```diff
@@ -1,9 +1,17 @@
+import {require as requireDefault} from "d3-require";
+
 async function remote_fetch(file) {
   const response = await fetch(await file.url());
   if (!response.ok) throw new Error(`Unable to load file: ${file.name}`);
   return response;
 }
 
+async function dsv(file, format, {array = false, type = null} = {}) {
+  const [text, d3] = await Promise.all([file.text(), requireDefault("d3-dsv@2.0.0/dist/d3-dsv.min.js")]);
+  if (type === true) type = d3.autoType;
+  return d3[`${format}Parse${array ? "Rows" : ""}`](text, type);
```

Comment:
This is a little scary since my brain went immediately to "user can specify an arbitrary function name?" but 1. both `format` and `array` aren't directly from the user, 2. it'll always be a Parse function, 3. it'll always be constrained to d3-dsv.

Reply:
It should be equivalent to:

```js
  return (format === "csv"
      ? (array ? d3.csvParseRows : d3.csvParse)
      : (array ? d3.tsvParseRows : d3.tsvParse))
    (text, map);
```

Would you prefer that?

Reply:
Yeah, I think so. It future-proofs it a bit from someone not understanding the implications when calling this function. It's probably more legible too!